tests/crop.py

```diff
@@ -0,0 +1,907 @@
+    return [-1, 0]
+
+
+@pytest.fixture
+def origin_zero():
+    return [0, 0]
+
+
+@pytest.fixture
+def origin_0_1():
+    return [0, 1]
+
+
+@pytest.fixture
+def shape_0x0():
+    return [0, 0]
+
+
+@pytest.fixture
+def indices_1_1():
+    return [1, 1]
+
+
+@pytest.fixture
+def indices_3_3():
+    return [3, 3]
+
+
+@pytest.fixture
+def tile_shape_2x2():
+    return [2, 2]
+
+
+@pytest.fixture
+def tile_subregion_2x2():
+    return [
+        [0, 0],
+        [2, 2]
+    ]
+
+
+@pytest.fixture
+def level0_shape_6x6():
+    return [6, 6]
+
+
+@pytest.fixture
+def scaled_shape_4x4():
+    return [4, 4]
+
+
+@pytest.fixture
+def level1_shape_3x3():
+    return [3, 3]
+
+
+@pytest.fixture
+def max_4():
+    return 4
+
+
+@pytest.fixture
+def num_levels_2():
+    return 2
+
+
+@pytest.fixture
+def round_up():
+    return True
+
+
+@pytest.fixture
+def round_down():
+    return False
+
+
+@pytest.fixture
+def real_tiles_red_mask(dirname):
+    return [
+        [
+            np.load(Path(dirname, 'data/red/0/0/tile.npy').resolve()),
+            np.load(Path(dirname, 'data/red/1/0/tile.npy').resolve()),
+            np.load(Path(dirname, 'data/red/2/0/tile.npy').resolve()),
+            np.load(Path(dirname, 'data/red/3/0/tile.npy').resolve())
+        ],
+        [
+            np.load(Path(dirname, 'data/red/0/1/tile.npy').resolve()),
+            np.load(Path(dirname, 'data/red/1/1/tile.npy').resolve()),
+            np.load(Path(dirname, 'data/red/2/1/tile.npy').resolve()),
+            np.load(Path(dirname, 'data/red/3/1/tile.npy').resolve())
+        ],
+        [
+            np.load(Path(dirname, 'data/red/0/2/tile.npy').resolve()),
+            np.load(Path(dirname, 'data/red/1/2/tile.npy').resolve()),
+            np.load(Path(dirname, 'data/red/2/2/tile.npy').resolve()),
+            np.load(Path(dirname, 'data/red/3/2/tile.npy').resolve())
+        ],
+        [
+            np.load(Path(dirname, 'data/red/0/3/tile.npy').resolve()),
+            np.load(Path(dirname, 'data/red/1/3/tile.npy').resolve()),
+            np.load(Path(dirname, 'data/red/2/3/tile.npy').resolve()),
+            np.load(Path(dirname, 'data/red/3/3/tile.npy').resolve())
+        ],
+    ]
+
+
+@pytest.fixture
+def tile_shape_1024x1024():
+    return [1024, 1024]
+
+
+@pytest.fixture
+def hd_shape_1920x1080():
+    return [1920, 1080]
+
+
+@pytest.fixture
+def hd_stitched(hd_shape_1920x1080, color_green):
+    hd_w, hd_h = hd_shape_1920x1080
+    return np.ones((hd_h, hd_w, 3)) * color_green
+
+
+@pytest.fixture
+def hd_tiles_green_mask():
+    return [
+        [
+            np.ones((1024, 1024)),
+            np.ones((1024, 896))
+        ],
+        [
+            np.ones((56, 1024)),
+            np.ones((56, 896))
+        ]
+    ]
+
+
+@pytest.fixture
+def tile_shape_256x256():
+    return [256, 256]
+
+
+@pytest.fixture
+def real_shape_1024x1024():
+    return [1024, 1024]
+
+
+@pytest.fixture
+def real_stitched_with_gamma(dirname):
```

Comment:
How big are the test images in S3? I think if possible it would be preferable not to need internet access to run the tests.